storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3MultiPartOutputStreamTest.java

```diff
@@ -135,102 +136,116 @@ void writesOneByte() throws Exception {
         verify(mockedS3).uploadPart(any(UploadPartRequest.class));
         verify(mockedS3).completeMultipartUpload(any(CompleteMultipartUploadRequest.class));
 
+        final UploadPartRequest value = uploadPartRequestCaptor.getValue();
         assertUploadPartRequest(
-            uploadPartRequestCaptor.getValue(),
+            value,
+            value.getInputStream().readAllBytes(),
             1,
             1,
             new byte[] {1});
         assertCompleteMultipartUploadRequest(
             completeMultipartUploadRequestCaptor.getValue(),
-            List.of(new PartETag(1, "SOME_ETAG"))
+            Map.of(1, "SOME_ETAG")
         );
     }
 
     @Test
     void writesMultipleMessages() throws Exception {
         final int bufferSize = 10;
-        final byte[] message = new byte[bufferSize];
 
         when(mockedS3.initiateMultipartUpload(any(InitiateMultipartUploadRequest.class)))
             .thenReturn(newInitiateMultipartUploadResult());
+
+        final Map<Integer, UploadPartRequest> uploadPartRequests = new ConcurrentHashMap<>();
+        final Map<Integer, byte[]> uploadPartContents = new ConcurrentHashMap<>();
         when(mockedS3.uploadPart(uploadPartRequestCaptor.capture()))
```

Comment:
This is weird. What are you mocking? Do we expect this method to answer when we called it with a captor? What will happen it will not be called and then we will try to get a captured value?

Reply:
And same for other places where captor is used

Reply:
Good catch, probably copy-pasted before. But you're right, captured values are not used on this cases and I'm replacing with `any(...)` instead